site/docs/recipes/0001-reusable-helpers.md

```diff
@@ -0,0 +1,234 @@
+
+See [this recipe](https://kysely.dev/docs/recipes/expressions) to learn more about expressions.
+
+So we've learned that everything's an expression and that expressions are composable. Let's put this idea to use:
+
+```ts
+const persons = await db
+  .selectFrom('person')
+  .select(['id', 'first_name'])
+  .where(({ eb, ref, val }) => eb(
+    concat(
+      lower(ref('first_name')),
+      val(' '),
+      upper(ref('last_name'))
+    ),
+    '=',
+    'sylvester STALLONE'
+  ))
+  .execute()
+```
+
+So far we've only used our helper functions in the first argument of `where` but you can use them anywhere:
+
+```ts
+const persons = await db
+  .selectFrom('person')
+  .innerJoin('pet', (join) => join.on(eb => eb(
+    'person.first_name', '=', lower(eb.ref('pet.name'))
+  )))
+  .select(({ ref, val }) => [
+    'first_name',
+    // If you use a helper in `select`, you need to always provide an explicit
+    // name for it using the `as` method.
+    concat(ref('person.first_name'), val(' '), ref('pet.name')).as('name_with_pet')
+  ])
+  .orderBy(({ ref }) => lower(ref('first_name')))
+  .execute()
+```
+
+## Reusable helpers using `ExpressionBuilder`
+
+Here's an example of a helper function that uses the expression builder instead of raw SQL:
+
+```ts
+import { Expression, expressionBuilder } from 'kysely'
+
+function idsOfPersonsThatHaveDogNamed(name: Expression<string>) {
+  const eb = expressionBuilder<DB>()
+
+  // A subquery that returns the identifiers of all persons
+  // that have a dog named `name`.
+  return eb
+    .selectFrom('pet')
+    .select('pet.owner_id')
+    .where('pet.species', '=', 'dog')
+    .where('pet.name', '=', name)
+}
+```
+
+And here's how you could use it:
+
+```ts
+const dogName = 'Doggo'
+
+const persons = await db
+  .selectFrom('person')
+  .selectAll('person')
+  .where((eb) => eb(
+    'person.id', 'in', idsOfPersonsThatHaveDogNamed(eb.val(dogName))
+  ))
+  .execute()
+```
+
+Note that `idsOfPersonsThatHaveDogNamed` doesn't execute a separate query but instead returns a subquery expression that's compiled as a part of the parent query:
+
+```sql
+select
+  person.*
+from
+  person
+where
+  person.id in (
+    select pet.owner_id
+    from pet
+    where pet.species = 'dog'
+    and pet.name = ?
+  )
+```
+
+In all our examples we've used the following syntax:
+
+```ts
+.where(eb => eb(left, operator, right))
+```
+
+When the expression builder `eb` is used as a function, it creates a binary expression. All binary expression with a comparison operator are represented as a `Expression<SqlBool>`. You don't always need to return `eb(left, operator, right)` from the callback though. Since `Expressions` are composable and reusable, you can return any `Expression<SqlBool>`.
```

Comment:
```suggestion
When the expression builder `eb` is used as a function, it creates a binary expression. All binary expressions with a comparison operator are represented as `Expression<SqlBool>`. You don't always need to return `eb(left, operator, right)` from the callback though. Since `Expressions` are composable and reusable, you can return any `Expression<SqlBool>`.
```